preprocessing_of_corpus.py

```diff
@@ -0,0 +1,150 @@
+    t = re.sub(r"($|€|¥|₹|£)","",t)
+    t = re.sub(r"(%)","",t)
+    t = re.sub(r"\d+","",t)
+    t = re.sub(r"\n","",t)
+    t = re.sub(r"\xa0", "", t)
+    return t
+
+#Return punctuations from text
+def pun(text):
+
+    table = str.maketrans("","", string.punctuation)
+    t = text.translate(table)
+    return t
+
+#loading the engish language for lemmatization
+nlp = spacy.load('en_core_web_sm')
+
+#reducing the words in a sentence to its root form
+def lemmatizer(text):
+    """Returns text after lemmatization"""
+    sent = []
+    doc = nlp(text)
+    for word in doc:
+        sent.append(word.lemma_)
+
+    return " ".join(sent)
+
+#Returns text after removing some extra symbols
+def extras(sentences):
+
+    t = re.sub(r"\"|\—|\'|\’\n","",sentences)
+    word_list = t.split()
+    for index, word in enumerate(word_list):
+        if len(word) <=1:
+            del word_list[index]
+    t = ' '.join(word_list)
+
+    return t
+
+# Stop words removal
+def stop_word(sentence):
+    list_ = []
+    stop_words = stopwords.words('english')
+    words_list = sentence.split()
+    for word in words_list:
+        if word not in stop_words:
+            list_.append(word)
+    return ' '.join(list_)
+#returns list of words after eliminating all the repeated words in the list
+def remove_repetative_words(words):
+
+    repeat_words=[]
+    for word in words:
+        if word not in repeat_words:
+            repeat_words.append(word)
+    return repeat_words
+
+#Extract sentences with the required word and inserts into the database
+def get_sentences_with_word(words, sentences):
+    sentences_with_word = []
+    for word in words:
+        for sentence in sentences:
+            words_in_sentence = word_tokenize(sentence)
+
+            for word_in_sentence in words_in_sentence:
+                if word_in_sentence == word :
+                    sentences_with_word.append(sentence)
+
+        x = insert_sentences(word, sentences_with_word)#function call for inserting the sentences into db
+        print("word:", word, "sentence:", sentences_with_word)
+        print(x)
+        sentences_with_word  = []
+
+
+
+def main():
+    """Main Function"""
+    #with open('GOT.txt', 'r') as f:
+     #   contents = f.readlines()
+
+    with codecs.open("text1.txt", "r", "utf-8") as fileT:
+        corpus_raw = fileT.read()
+
+    # Joining the lines to make text block
+    #contents = ''.join(contents)
+
+    print("Starting to preprocess file")
+
+    # Sentence tokenize the text
+    sent_tokenized = sentences(corpus_raw)
+
+    # Removing all the unnecessary things from the text
+    get_sentences_without_special_symbols = [remove(line) for line in sent_tokenized]
+
+    # Removing punctuations
+    get_sentences_without_punctuations = [pun(line.lower()) for line in get_sentences_without_special_symbols]
+
+    # Removing extras from the sentences
+    get_sentences_without_extras = [extras(sent) for sent in get_sentences_without_punctuations]

```

Comment:
What is considered extras here?

Reply:
After replacing the unnecessary symbols with spaces there might be series of gaps/spaces. The extras can remove such spaces between.